File: src/main/java/com/company/payroll/mapper/LoanMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Loan;

@Repository
public interface LoanMapper {
    int deleteByPrimaryKey(Integer loanId);

    int insert(Loan row);

    int insertSelective(Loan row);
    
    List<Loan> selectList();
    
    List<Loan> selectListByEId(int eid);

    Loan selectByPrimaryKey(Integer loanId);

    int updateByPrimaryKeySelective(Loan row);

    int updateByPrimaryKey(Loan row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/LoanMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Loan
import org.springframework.stereotype.Repository

@Repository
interface LoanMapper {
  fun deleteByPrimaryKey(loanId: Int): Int
  fun insert(row: Loan): Int
  fun insertSelective(row: Loan): Int
  fun selectList(): List<Loan>?
  fun selectListByEId(staffId: Int): List<Loan>?
  fun selectByPrimaryKey(loanId: Int): Loan?
  fun updateByPrimaryKeySelective(row: Loan): Int
  fun updateByPrimaryKey(row: Loan): Int
}